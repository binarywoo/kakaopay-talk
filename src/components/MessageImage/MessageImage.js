import React, { useState, useCallback, useEffect } from 'react'
import ImageModal from '../ImageModal/index'
import { scrollToBottom } from '../../libs/chatAppUtils'

const MessageImage = ({ src }) => {
  const [showImageModal, setShowImageModal] = useState(false)
  const [isLoaded, setIsLoaded] = useState(false)

  useEffect(() => {
    if (isLoaded) scrollToBottom()
  }, [isLoaded])

  const onClickImg = useCallback(() => {
    setShowImageModal(true)
  }, [src])

  return (
    <div>
      <img
        style={{ maxWidth: 200, cursor: 'pointer' }}
        src={src}
        alt={src}
        onClick={onClickImg}
        onLoad={() => setIsLoaded(true)}
      />
      <ImageModal
        visible={showImageModal}
        title={null}
        img={src}
        onClose={() => setShowImageModal(false)}
        alt={src}
      />
    </div>
  )
}

export default MessageImage
