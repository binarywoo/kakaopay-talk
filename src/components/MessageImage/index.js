import MessageImage from './MessageImage'

export default MessageImage
