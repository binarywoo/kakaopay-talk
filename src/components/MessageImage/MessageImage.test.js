import React from 'react'
import { shallow, mount } from 'enzyme'
import MessageImage from './MessageImage'

describe('MessageImage', () => {
  let component = null
  const dummyImage =
    'https://firebasestorage.googleapis.com/v0/b/hand-pokemon-2.appspot.com/o/workshopImages%2FXL6bftAQxqyLPFSStXcdAnW9aoPFncOW?alt=media&token=36cca2b9-0591-4ff3-b27f-0137a2126808'
  console.log(MessageImage)
  it('정상적으로 렌더링이 된다.', () => {
    component = shallow(<MessageImage src={dummyImage} />)
  })

  it('스냅샷과 매칭이 된다.', () => {
    expect(component).toMatchSnapshot()
  })
})
