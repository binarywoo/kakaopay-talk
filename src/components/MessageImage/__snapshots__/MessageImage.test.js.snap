// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageImage 스냅샷과 매칭이 된다. 1`] = `
<div>
  <img
    alt="https://firebasestorage.googleapis.com/v0/b/hand-pokemon-2.appspot.com/o/workshopImages%2FXL6bftAQxqyLPFSStXcdAnW9aoPFncOW?alt=media&token=36cca2b9-0591-4ff3-b27f-0137a2126808"
    onClick={[Function]}
    onLoad={[Function]}
    src="https://firebasestorage.googleapis.com/v0/b/hand-pokemon-2.appspot.com/o/workshopImages%2FXL6bftAQxqyLPFSStXcdAnW9aoPFncOW?alt=media&token=36cca2b9-0591-4ff3-b27f-0137a2126808"
    style={
      Object {
        "cursor": "pointer",
        "maxWidth": 200,
      }
    }
  />
  <MyComponent
    alt="https://firebasestorage.googleapis.com/v0/b/hand-pokemon-2.appspot.com/o/workshopImages%2FXL6bftAQxqyLPFSStXcdAnW9aoPFncOW?alt=media&token=36cca2b9-0591-4ff3-b27f-0137a2126808"
    img="https://firebasestorage.googleapis.com/v0/b/hand-pokemon-2.appspot.com/o/workshopImages%2FXL6bftAQxqyLPFSStXcdAnW9aoPFncOW?alt=media&token=36cca2b9-0591-4ff3-b27f-0137a2126808"
    onClose={[Function]}
    title={null}
    visible={false}
  />
</div>
`;
